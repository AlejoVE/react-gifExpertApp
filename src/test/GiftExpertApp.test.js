import React from 'react';
import { shallow } from "enzyme";
import GiftExpertApp from '../GiftExpertApp';
import '@testing-library/jest-dom';

describe('tests in <GiftExpertApp /> component', () => {

    test('should show <GiftExpertApp /> component correctly ', () => {
        let wrapper = shallow(<GiftExpertApp />)
        
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should  show a list of categories', () => {
        
        const categories = ['Marvel', 'DC'];
        
        let wrapper = shallow(<GiftExpertApp  defaultCategories={categories}/>)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })
    
    
})
