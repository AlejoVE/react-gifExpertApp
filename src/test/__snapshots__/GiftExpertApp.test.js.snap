// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <GiftExpertApp /> component should  show a list of categories 1`] = `
<Fragment>
  <h2>
    GitExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Marvel"
      key="Marvel"
    />
    <GifGrid
      category="DC"
      key="DC"
    />
  </ol>
</Fragment>
`;

exports[`tests in <GiftExpertApp /> component should show <GiftExpertApp /> component correctly  1`] = `
<Fragment>
  <h2>
    GitExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
